tools/wpt/revlist.py

```diff
@@ -0,0 +1,128 @@
+import argparse
+import calendar
+import logging
+import os
+import time
+from tools.wpt.testfiles import get_git_cmd
+
+here = os.path.dirname(__file__)
+wpt_root = os.path.abspath(os.path.join(here, os.pardir, os.pardir))
+
+logger = logging.getLogger()
+
+MYPY = False
+if MYPY:
+    # MYPY is set to True when run under Mypy.
+    from typing import Any
+    from typing import Dict
+    from typing import List
+    from typing import Text
+
+
+def calculate_min_timestamp(until, epoch, offset):
+    return ((((until - offset) // epoch)) * epoch) + offset
+
+
+def parse_epoch(value):
```

Comment:
In https://docs.python.org/2/library/argparse.html#type it looks like it's possible to provide a custom value parser. You could use `type=parse_epoch` in `parser.add_argument` to avoid also doing `epoch = parse_epoch(kwargs["epoch"])`.

As an optional simplification, I think this could also be simplified into something without hard coded epoch times, leaving that to the code that maintains the branches. It'd be shorter:
```py
def epoch_spec(string):
    UNIT_DICT = {"h": 3600, "d": 86400, "w": 604800}
    base = string[:-1]
    unit = string[-1:]
    if base.isdigit() and unit in UNIT_DICT:
        return int(base) * UNIT_DICT[unit]
    raise argparse.ArgumentTypeError('must be digits followed by h/d/w')
```

Then we use `--epoch=6h`, `--epoch=1d` and `--epoch=1w`.